File: python/conway.py
class conway:

    size = 20

    grid = [[0 for col in range(size)] for row in range(size)]
    nextgen = [[0 for col in range(size)] for row in range(size)]

    def __init__(self, size):
        self.size = size

        # blinker seed
        self.grid[2][3] = 1
        self.grid[2][4] = 1
        self.grid[2][5] = 1

        # glider seed
        self.grid[8][8] = 1
        self.grid[9][9] = 1
        self.grid[10][9] = 1
        self.grid[10][8] = 1
        self.grid[10][7] = 1



    def getGrid(self):
        return self.grid


    def iterate(self):

        for col in range(self.size):
            for row in range(self.size):
                neighbors = self.getNeighbors(col, row)

                if self.grid[col][row] == 1:
                    if  neighbors < 2:
                        self.nextgen[col][row] = 0
                    elif neighbors > 3:
                        self.nextgen[col][row] = 0
                    else:
                        self.nextgen[col][row] = self.grid[col][row]
                elif self.grid[col][row] == 0:
                    if neighbors == 3:
                        self.nextgen[col][row] = 1
                    else:
                        self.nextgen[col][row] = self.grid[col][row]


    def getNeighbors(self, col, row):
        count = 0
        for i in range(col-1, col+2):
            for j in range(row-1, row+2):
                if i == col and j == row:
                    pass
                else:
                    if not self.outOfBounds(i,j):
                        if self.grid[i][j] == 1:
                            count = count + 1
        return count


    def outOfBounds(self, x, y):
        if x==-1 or x == self.size or y == -1 or y == self.size:
            return True
        return False

    def Update(self):
        temp = self.nextgen
        self.nextgen = self.grid
        self.grid = temp

    def takeAStep(self):
        self.iterate()
        self.Update()
        return self.getGrid()

File: js/conway.js
conway = {

    _size: 16,

    world: null,

    init: function() {
        conway.world = conway.getEmptyWorld(conway._size);

        conway.world[8][8] = 1;
        conway.world[9][9] = 1;
        conway.world[10][9] = 1;
        conway.world[10][8] = 1;
        conway.world[10][7] = 1;

    },

    iterate: function() {
        var newWorld = conway.getEmptyWorld(conway._size);

        for (x = 0; x < conway._size; x++) {
            for (y = 0; y < conway._size; y++) {
                var neighborCount = conway.getAliveNeighborCount(x, y);

                if (conway.world[x][y] == 0) {
                    if (neighborCount == 3) {
                        newWorld[x][y] = 1;
                    } else {
                        newWorld[x][y] = conway.world[x][y];
                    }
                } else {
                    if (neighborCount < 2) {
                        newWorld[x][y] = 0;
                    } else if (neighborCount > 3) {
                        newWorld[x][y] = 0;
                    } else {
                        newWorld[x][y] = conway.world[x][y];
                    }
                }
            }
        }
        conway.world = newWorld;
        return conway.world;
    },

    isOutOfBounds: function(x, y) {
        if (x >= conway._size || x < 0 || y >= conway._size || y < 0) {
            return true;
        }
        return false;
    },

    getAliveNeighborCount: function(x, y) {
        var count = 0;
        for (i = -1; i < 2; i++) {
            for (j = -1; j < 2; j++) {
                if (!conway.isOutOfBounds(x + i, y + j)) {
                    if (i == 0 && j == 0) {

                    } else {
                        if (conway.world[x + i][y + j] == 1) {
                            count++;
                        }
                    }
                }
            }
        }
        return count;
    },

    getEmptyWorld: function(size) {
        var _w = [];
        for (y = 0; y < size; y++) {
            _w[y] = [];

            for (x = 0; x < size; x++) {
                _w[y][x] = 0;
            }
        }
        return _w;
    }
}